feat(utxo-lib): add network configuration for DOGE

DWJ01 opened the pull request:
For now we cannot make dogecoin tx with an unspent greater than Number.MAX_SAFE_INTEGER. We will add support for this in the future

BG-51068

williamqin-bitgo reviewed (changes requested):
Need changes to pass unit tests - check out https://github.com/BitGo/BitGoJS/compare/master...BG-49864
(I'm unsure what the dependency audit test fail is though)

mmcshinsky-bitgo commented:
@DWJ01 Looks like you have lint errors failing your PR.

williamqin-bitgo commented on modules/utxo-lib/test/fixtures_thirdparty/nodes/dogecoin-1.14.5/src/test/data/sighash.json:
```diff
@@ -0,0 +1,503 @@
+[
+	["raw_transaction, script, input_index, hashType, signature_hash (result)"],
```
out of curiosity, how did you generate this file?

DWJ01 replied:
It's from [dogecoin github repo](https://github.com/dogecoin/dogecoin/blob/master/src/test/data/sighash.json).

williamqin-bitgo replied:
!!! I didn't know that's where you find it - awesome!

williamqin-bitgo commented on modules/utxo-lib/test/fixtures_thirdparty/nodes/dogecoin-1.14.5/src/test/data/tx_valid.json:
```diff
@@ -0,0 +1,514 @@
+[
+["The following are deserialized transactions which are valid."],
```
same as before, how did you generate this one?

DWJ01 replied:
It's also from [dogecoin github repo](https://github.com/dogecoin/dogecoin/blob/master/src/test/data/tx_valid.json)

williamqin-bitgo commented on modules/utxo-lib/test/networks.ts:
```diff
@@ -41,7 +42,13 @@ describe('networks', function () {
 
   const bitcoinTestnetSharedScriptPrefix = (network) =>
-    bitcoinTestnetSharedPubkeyPrefix(network) && !isLitecoin(network);
+    // bitcoinTestnetSharedPubkeyPrefix(network) && !isLitecoin(network);
```
I like this, remove the comment though

DWJ01 replied:
removed

williamqin-bitgo commented on modules/utxo-lib/test/networks.ts:
```diff
@@ -41,7 +42,13 @@ describe('networks', function () {
+      isBitcoinCash(network) ||
+      isBitcoinGold(network) ||
+      isBitcoinSV(network) ||
+      isDogecoin(network));
```
while you're at it, maybe make them the same order as `bitcoinTestnetSharedPubkeyPrefix`?

DWJ01 replied:
changed the order in `bitcoinTestnetSharedPubkeyPrefix` to meet correct alphabetical order.

williamqin-bitgo reviewed (dismissed):
looks good! squash commits before merging (unless that's done automatically)

OttoAllmendinger commented on modules/utxo-lib/test/integration_local_rpc/generate/outputScripts.util.ts:
```diff
@@ -88,7 +88,7 @@ export function createSpendTransactionFromPrevOutputs<T extends UtxoTransaction>
 
   const inputSum = prevOutputs.reduce((sum, { value }) => sum + value, 0);
-  const fee = 1000;
+  const fee = network === utxolib.networks.dogecoinTest ? 1000000 : 1000;
```
```suggestion
  const fee = network === utxolib.networks.dogecoinTest ? 1_000_000 : 1_000;
```

OttoAllmendinger reviewed (changes requested):
Please rebase on top of `master` and squash your commits.

Explain the current limits of dogecoin support in the commit message.

OttoAllmendinger reviewed (changes requested):
see comment

DWJ01 committed:
feat(utxo-lib): add network configuration for DOGE

For now we cannot make dogecoin tx with an unspent greater than Number.MAX_SAFE_INTEGER.

BG-51068
